Large root folder improvements

sberan opened the pull request:
Hi! I like this plugin, thank you for making it!

I noticed that the plugin was very slow to start up in my "Code" folder. There are a _ton_ of projects in this folder, many of them are not git projects. I figured there may be other people in the same boat as me, and it would be great for them to have a good experience when first trying out the plugin.

To that end, I've made the directory listing async run asynchronously, which stops the UI from locking up for large projects.

Additionally, I've added a "max depth" option which puts a cap on the number of directories to traverse. It's defaulted to 5, but can be increased for more extreme circumstances.

These two features result in a much better UX for me when first launching the plugin with a large root folder, I hope you agree!

Thanks,
Sam

geowarin commented:
:+1:  I was looking at the code just now and wanted to do something similar.
Thanks @sberan. Can't wait for it to get merged :)

prrrnd commented:
Hi @sberan! Thank you for your work, that's awesome!
Making it asynchronously was in the back on my mind, but I just had no time to work on this package.
The max depth option sounds pretty smart too! Would you like to update the README or do you want me to do it?

sberan commented:
@prrrnd thanks for taking a look at this! Time is a scarce resource ;)

I've updated the README.md and rebased, the CI build is now passing.

I'm really enjoying this plugin now, looking forward to getting this merged and getting a release out!

sberan commented:
update, @prrrnd : hold off on merging this one, I think I found a bug.

sberan commented:
ok @prrrnd - this is ready to go!

prrrnd commented:
Thanks for the awesome work @sberan ! I'll merge and release today :)

sberan commented:
Right on! Thanks for getting a release out so fast!

:ship: :trophy:
